packages/hash/frontend/src/shared/layout/plain-layout.tsx

```diff
@@ -22,7 +23,7 @@ export const PlainLayout: FunctionComponent<{
         options={{ showSpinner: false }}
         showOnShallow
       />
-      {children}
+      <EntityTypesContextProvider>{children}</EntityTypesContextProvider>
```

Comment:
Short story: you're right, it should be in `_app.page.tsx` and I'll move it.

Long story: The reason I made this a global provider is because it's used in the sidebar. So I only put it in `LayoutWithSidebar` as it's not necessarily required to fetch entity types on pages which don't have a sidebar (i.e, the login page), unless the page itself requests it.

My first attempt at doing this had a sort of 'baton' effect, where if `EntityTypesContextProvider` was nested inside of another `EntityTypesContextProvider`, instead of fetching entities again and providing them again, it would just render its children, meaning components would subscribe to the top level context provider. The idea was that pages that use entity types are still wrapped in a provider, but it just detects that entity types have already been fetched and doesn't refetch them. This would mean you could safely change pages between a layout with a sidebar and without, and data fetching would still happen as expected, and additionally entity types wouldn't be fetched on pages that didn't need them.

This felt overly complex though, so I just added an `ensureFetched` method to the context value, and then made it so the hook subscribing to the context called it in an effect. I also removed the automatic fetch, so now entity types aren't fetched until the first time `ensureFetched` is called. This meant it was safe to move the provider up to the top level, without it leading to entity types being fetched when they weren't used on the page.

When I did that, I just walked the dependency tree to find the highest level, without really thinking I was in a layout, which was silly of me. I'll fix that.

Reply:
Thanks for the detailed explanation!